### PR TITLE
port framework_agreement_requisition

### PR DESCRIPTION
**Pull request by lepistone**
- [x] make tests pass in v8
- [x] functional test

### CODE REVIEW

**Comment by lepistone** [comment]
tests are red because of https://github.com/OCA/purchase-workflow/pull/32

**Comment by yvaucher** [comment]
:+1:

**Comment by lepistone** [comment]
Hi @gurneyalex @jgrandguillaume I marked as WIP because that probably needs some functional testing. Please restore "needs review" and update the checklist when it's OK for you.

Thanks!

**Comment by lepistone** [comment]
I actually have some work to do here. Let's leave it WIP for now.

**Comment by lepistone** [comment]
I did some manual testing. The agreement is generated with price zero. Being that a missing feature, I added doc and a popup to tell the user that prices have to be filled up manually.

**Comment by lepistone** [comment]
I am adding a few minor things. What is there should not change significantly.

**Comment by lepistone** [comment]
Minor change done.

File: framework_agreement_requisition/model/purchase_requisition.py [hunk 1]
```diff
@@ -58,9 +84,8 @@ def select_agreement(self, cr, uid, agr_id, context=None):
         if isinstance(agr_id, (list, tuple)):
             assert len(agr_id) == 1
             agr_id = agr_id[0]
-        wf_service = netsvc.LocalService("workflow")
-        return wf_service.trg_validate(uid, 'purchase.requisition',
-                                       agr_id, 'select_agreement', cr)
+        return workflow.trg_validate(uid, 'purchase.requisition',
+                                     agr_id, 'select_agreement', cr)
```
Comment (yvaucher) [inline framework_agreement_requisition/model/purchase_requisition.py]: You can use

```
class BaseModel(object):
    def signal_workflow(self, cr, uid, ids, signal, context=None):
```

It also works for multiple ids.

**Comment by yvaucher** [comment]
it misses the pot update for confirm_generate_agreement wizard

**Comment by lepistone** [comment]
@yvaucher done, thanks!

**Comment by yvaucher** [comment]
:+1: @lepistone thanks

**Comment by lepistone** [comment]
Applied @yvaucher's remark to another file, too.

**Comment by gurneyalex** [comment]
:+1: merging!

**Pull request merged**